Update marzban.sh

DigneZzZ opened the pull request:
I'm significantly improved the original Marzban installation script.

## Key Changes:

1. **Added MySQL Support**:
   - You can now install Marzban using MySQL by specifying `--database mysql` during installation.
   - The script generates a `docker-compose.yml` with MySQL configuration and updates the `.env` file accordingly.

2. **Enhanced Xray-core Version Selection Interface**:
   - The `get_xray_core` function now features a user-friendly, colored menu.
   - Select from available Xray-core versions or manually enter a specific version.
   - Improved validation ensures only valid versions are accepted.

3. **Suppressed Logs During Installation and Execution**:
   - To improve readability, unnecessary logs are now hidden during installation and command execution.
   - Essential success and error messages are still displayed for clarity.

## Brief Usage Instructions:

- **Install Marzban with MySQL**:

  ```bash
  sudo bash -c "$(curl -sL https://github.com/Gozargah/Marzban-scripts/raw/master/marzban.sh)" @ install --database mysql
  ```

- **Install Marzban with MariaDB**:

  ```bash
  sudo bash -c "$(curl -sL https://github.com/Gozargah/Marzban-scripts/raw/master/marzban.sh)" @ install --database mariadb
  ```
  
  - **Install Marzban with MariaDB and Dev branch**:

  ```bash
  sudo bash -c "$(curl -sL https://github.com/Gozargah/Marzban-scripts/raw/master/marzban.sh)" @ install --database mariadb --dev
  ```

  - **Install Marzban with MariaDB and Manual version **:

  ```bash
  sudo bash -c "$(curl -sL https://github.com/Gozargah/Marzban-scripts/raw/master/marzban.sh)" @ install --database mariadb --version v0.5.2
  ```

- **Update or Change Xray-core Version**:

  ```bash
  sudo marzban core-update
  ```

DigneZzZ commented:
### Marzban-node.sh:
I fixed the script hanging issue during package installation by removing background execution and output redirection in the get_xray_core function. I also modified the install_package function to suppress installation logs using quiet mode flags and output redirection, while keeping errors visible for debugging.

DigneZzZ commented:
Enhanced the `usage` function in Marzban and Marzban-node CLI for improved readability and functionality.

- **Marzban**:
  - Added new editing options:
    - `edit`: Opens `docker-compose.yml` for editing.
    - `edit-env`: Opens `.env` for editing.
  - Improved help display with color-coded sections and structured command descriptions for better readability.
  
- **Marzban-node**: Enhanced visual layout of the help menu for easier navigation and readability.

These updates improve the user experience by adding new functionality and making CLI instructions clearer and more visually organized.

iamtheted commented:
I really appreciate you!
This will make things easier for sure.

DigneZzZ commented:
Some improvements from [the PR](https://github.com/Gozargah/Marzban-scripts/pull/32) have been added.

MILADALEF commented:
بررسی نشد ؟؟

zapzocket commented:
tnx!!!!!!!!!!!!!

Aknuun commented:
عجیبه که چرا pr های به درد بخور دیر تایید میشه
اصلا سینت انگار ول کرده رفته

Multivpnlive commented:
امیدوارم زودتر بررسی بشه این pr

Iman-GTR commented:
I hope so Marzban check and add this PR 🙏 as soon as possible thanks to Marzban🌹

nemppn commented:
خیلی کاربردی هست امیدوارم زودتر بیاد رو پنل ،✌️😎

nemppn commented:
I hope so Marzban check and add this PR 🙏 as soon as possible Thanks to Marzban
Thanks to Alexey 😎✌️
you are best brooo

Надеюсь, Marzban проверит и добавит этот PR 🙏 как можно скорее Спасибо Marzban
Спасибо Алексею 😎✌️
Ты лучший, братан

Nima786 commented:
این خیلی کاربردی و مفید هست و امیدوارم زودتر مرج بشه 👍

iamtheted commented:
After testing the script, we found out some points:

1.MySQL username is set to marzban while the default is root and that's what backup scripts use.

2.There's no PhpMyAdmin included, The default password is set to the word "password" so if user decides to setup PhpMyAdmin, he MUST change the password and if he forgets the default password will be the word "password". I believe the script must ask for a password in the first place, choosing a default one is not reasonable.

3.MySQL version is set to 8.3 while it has to be latest 

I believe we need to include the default docker file in marzban's [document](https://gozargah.github.io/marzban/fa/examples/mysql) and only add a step in the script to ask for a password. no need for the extra stuff which makes it super complex.

DigneZzZ commented:
> After testing the script, we found out some points:
> 
> 1.MySQL username is set to marzban while the default is root and that's what backup scripts use.
> 
> 2.There's no PhpMyAdmin included, The default password is set to the word "password" so if user decides to setup PhpMyAdmin, he MUST change the password and if he forgets the default password will be the word "password". I believe the script must ask for a password in the first place, choosing a default one is not reasonable.
> 
> 3.MySQL version is set to 8.3 while it has to be latest
> 
> I believe we need to include the default docker file in marzban's [document](https://gozargah.github.io/marzban/fa/examples/mysql) and only add a step in the script to ask for a password. no need for the extra stuff which makes it super complex.

The message above doesn’t quite match reality.

But I’ll make some adjustments.

DigneZzZ commented:
### Changelog

**Implemented Adjustments:**

- Optimized logging for MySQL/MariaDB.
- MySQL ROOT password is now always generated automatically.
- User is now prompted to set a password for the `marzban` user.
  - If no password is set for the `marzban` user, one will be generated automatically.
- All database variables are now saved in the `.env` file.
- Minor code improvements.

Nima786 commented:
I ran the script with the first command (installed marzban with mysql) and gave this error on Ubuntu 24.04 and 22.04.
"Invalid version format. Please enter a valid version (e.g. v0.5.2)"

Nima786 commented:
> > I ran the script with the first command (installed marzban with mysql) and gave this error on Ubuntu 24.04 and 22.04.
> > "Invalid version format. Please enter a valid version (e.g. v0.5.2)"
> > Then I specified the version as well, but the same error.
> 
> Which command?

sudo bash -c "$(curl -sL https://github.com/Gozargah/Marzban-scripts/raw/master/marzban.sh)" @ install --database mysql

DigneZzZ commented:
> > > I ran the script with the first command (installed marzban with mysql) and gave this error on Ubuntu 24.04 and 22.04.
> 
> > > "Invalid version format. Please enter a valid version (e.g. v0.5.2)"
> 
> > > Then I specified the version as well, but the same error.
> 
> > 
> 
> > Which command?
> 
> 
> 
> sudo bash -c "$(curl -sL https://github.com/Gozargah/Marzban-scripts/raw/master/marzban.sh)" @ install --database mysql

No wonder, this PR hasn’t been merged yet, so this command can’t work.

Nima786 commented:
> > > > I ran the script with the first command (installed marzban with mysql) and gave this error on Ubuntu 24.04 and 22.04.
> > 
> > 
> > > > "Invalid version format. Please enter a valid version (e.g. v0.5.2)"
> > 
> > 
> > > > Then I specified the version as well, but the same error.
> > 
> > 
> > > 
> > 
> > 
> > > Which command?
> > 
> > 
> > sudo bash -c "$(curl -sL https://github.com/Gozargah/Marzban-scripts/raw/master/marzban.sh)" @ install --database mysql
> 
> No wonder, this PR hasn’t been merged yet, so this command can’t work.

Oh thanks, is there any way we can test it?

DigneZzZ commented:
> > > > > I ran the script with the first command (installed marzban with mysql) and gave this error on Ubuntu 24.04 and 22.04.
> > > 
> > > 
> > > > > "Invalid version format. Please enter a valid version (e.g. v0.5.2)"
> > > 
> > > 
> > > > > Then I specified the version as well, but the same error.
> > > 
> > > 
> > > > 
> > > 
> > > 
> > > > Which command?
> > > 
> > > 
> > > sudo bash -c "$(curl -sL https://github.com/Gozargah/Marzban-scripts/raw/master/marzban.sh)" @ install --database mysql
> > 
> > 
> > No wonder, this PR hasn’t been merged yet, so this command can’t work.
> 
> Oh thanks, is there any way we can test it?
yes, test it from here commands
https://github.com/DigneZzZ/Marzban-scripts-beta

Nima786 commented:
I tested and found that phpMyAdmin is not getting installed which I believe should be the part of the instructions.

DigneZzZ commented:
> I tested and found that phpMyAdmin is not getting installed which I believe should be the part of the instructions. Thank

I don’t agree with the practice of installing PMA directly with the script, as it creates an additional security risk for the server.

Nima786 commented:
> > I tested and found that phpMyAdmin is not getting installed which I believe should be the part of the instructions. Thank
> 
> I don’t agree with the practice of installing PMA directly with the script, as it creates an additional security risk for the server.

Yeah, it makes sense but if you make it optimal it would be nice, so users have choice.
But anyway, thanks for your efforts and really appreciate your contribution

iamtheted commented:
Everything is perfect! 
Thanks a lot!

Except for the docker sample for MySQL, which is still set to 8.3, shouldn't be latest or lts ?

DigneZzZ commented:
> Except for the docker sample for MySQL, which is still set to 8.3, shouldn't be latest or lts ?

Because that’s the decision I made.
Version 8.3 is tried and true. Many people still use it to this day. The latest version, 9.0+, can always be updated individually later if desired.

DigneZzZ commented:
### Marzban-node script updated:

- **Added display of the current installed Xray-core version during core installation and when calling help.**

  - Introduced a new function `get_current_xray_core_version()` to retrieve the installed Xray-core version.
  - Updated the `get_xray_core()` function to display the current core version prominently.
  - Modified the `usage` (help) function to include the current core version in the help output.

![image](https://github.com/user-attachments/assets/f2cfddb6-a2d2-478e-899f-fc9259620bcc)
![image](https://github.com/user-attachments/assets/314a5de7-c1fc-43df-8e1e-4f83ae4f41fe)

DigneZzZ commented:
- **Modified `get_current_xray_core_version()` to display the version from the Docker container if a custom Xray-core is not installed locally.**
  
  - If the local custom Xray binary is not found, the script now retrieves the Xray-core version from the running Docker container.
  - The version obtained from the container is displayed with the note "(in container)" to indicate its source.
  - This update ensures that users are always informed of the current Xray-core version, even when a custom version isn't installed.

Help menu:
![image](https://github.com/user-attachments/assets/96c486da-b8df-497a-906e-b0e6676aa116)

`marzban-node core-update` menu:
![image](https://github.com/user-attachments/assets/2b55bc51-9bf9-4a2f-ae14-07e6ad89b64d)
